components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -747,6 +747,20 @@ public String createApplication(ApplicationModel applicationModel, String templa
             validateCORSOrigins(applicationModel.getInboundProtocolConfiguration().getOidc().getAllowedOrigins());
         }
 
+        /*
+         * Validate the useExternalConsentPage property.
+         * We are only allowed to use external consent page for OIDC applications.
+         */
+        if (applicationModel.getInboundProtocolConfiguration() != null &&
+                applicationModel.getInboundProtocolConfiguration().getSaml() != null) {
+            if (applicationModel.getAdvancedConfigurations() != null && applicationModel.getAdvancedConfigurations()
+                    .getUseExternalConsentPage() != null &&
+                    applicationModel.getAdvancedConfigurations().getUseExternalConsentPage()) {
+                throw buildBadRequestError("Use external consent page is not supported for SAML " +
```

Comment:
Change the proper error message

Reply:
Resolved by https://github.com/wso2/identity-api-server/pull/445/commits/bad947186c79630ef5c172840e126dbc19878e12